components/Chat/ChatInput.vue

```diff
@@ -347,9 +357,41 @@
         }
       },
 
-      // Defer some load events til user interacts with chat
-      async onFocus () {
-        await this.updateEmoteList();
+      updateEmoteList() {
+        this.updateEmoteListInStore();
+
+        if( this.emoteList && this.emoteLinkMap && this.emoteLinkMap.size === this.emoteList.length ) {
+          console.log( 'Skipping emoteList update.' );
+          return;
+        }
+
+        if( !this.emoteLinkMap ) {
+          this.emoteLinkMap = new Map();
+        }
+
+        for( const emote of this.emoteList ) {
+          this.emoteLinkMap.set( emote.image, emote.value );
+        }
+      },
+
+      async onDrop( event ) {
+        if( !this.emoteLinkMap || this.emoteLinkMap.size < 1 ) {
+          await this.updateEmoteList();
+        }
+
+        const droppedText = event.dataTransfer.getData( "text/plain" );
+        const isEmoteLink = droppedText.startsWith( "https://cdn.bitwave.tv/static/emotes/" )
+                         || droppedText.startsWith( "https://cdn.bitwave.tv/uploads/" );
+        if( isEmoteLink ) {
+          const emoteLink = droppedText.replace( /\?[0-9]*$/g, '' );
+          const emote = this.emoteLinkMap.get( emoteLink );
+
+          event.preventDefault();
+
+          // In case the emote isn't found, ensure the link is pasted
+          if( emote ) { this.appendToMessage( emote ); }
+          else { this.appendToMessage( emoteLink ); }
```

Comment:
This `else` and the previous `preventDefault()`, as far as I understand it, shouldn't be necessary. However, links that get matched, but aren't emotes, sometimes don't get pasted.

Reply:
Even more of an issue was that the function call is `appendToChatMessage`. 

other than that, it only kicks in on drop like you said, so shouldn't be an issue any users manages to encounter on the day to day I imagine.